engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ImportExportScope.java

```diff
@@ -0,0 +1,70 @@
+package org.enso.interpreter.runtime.scope;
+
+import java.util.List;
+import org.enso.compiler.context.CompilerContext;
+import org.enso.interpreter.runtime.Module;
+import org.enso.interpreter.runtime.callable.function.Function;
+import org.enso.interpreter.runtime.data.EnsoObject;
+import org.enso.interpreter.runtime.data.Type;
+
+/**
+ * A proxy scope delegating to the underlying module's scope. Additionally, `ImportExportScope` may
+ * limit the number of types that are imported/exported.
+ */
+public class ImportExportScope implements EnsoObject {
```

Comment:
Why don't this object extend `EnsoObject`? It shouldn't flow thru user data right?

Make `final`.

Reply:
? `EnsoObject` is an interface, I don't understand the request.

Reply:
Yes, `EnsoObject` is an interface with a _purpose_ (described in its javadoc) of:
```
/** All non-primitive Enso types extends from {@code EnsoObject}. */
```
why do you believe `ImportExportScope` should be a **non-primitive Enso type**? Do you believe this object shall _flow thru user program_? Unless I am mistaken you have removed exposure of `ModuleScope` from `Meta.enso` claiming _"it is dangerous"_ - why do you believe `ImportExportScope` should be exposed (and manipulated) by Enso programs?

```suggestion
public final class ImportExportScope {
```